commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleDrive.scala

```diff
@@ -8,55 +8,84 @@ import squants.motion.AngularVelocity
+    * @return the parent's equivalent signal
+    */
+  protected def convertUnicycleToDrive(uni: UnicycleSignal): DriveSignal
 
-  private def toOpenDrive(unicycle: Signal[UnicycleSignal]): Signal[DriveSignal] = {
-    unicycle.map(convertUnicycleToDrive)
+  /**
+    * Uses the parent's open loop control for the equivalent drive signal for the unicycle signal
+    * @param unicycle the unicycle signal to drive with
+    * @return a signal controlled with open-loop on the parent
+    */
+  private def parentOpenLoop(unicycle: SignalLike[UnicycleSignal]): PeriodicSignal[DriveSignal] = {
+    unicycle.map(convertUnicycleToDrive).toPeriodic
   }
 
-  private def toClosedDrive(unicycle: SignalLike[UnicycleSignal]): PeriodicSignal[DriveSignal] = {
+  /**
+    * Uses the parent's closed loop control for the drive signal for the unicycle signal
+    * @param unicycle the unicycle signal to closed-loop drive with
+    * @return a signal controlled with closed-loop on the parent
+    */
+  private def parentClosedLoop(unicycle: SignalLike[UnicycleSignal]): PeriodicSignal[DriveSignal] = {
     driveClosedLoop(unicycle.map(convertUnicycleToDrive))
   }
 
   object UnicycleControllers {
-    def forwardOpen(forwardSpeed: Signal[Dimensionless]): PeriodicSignal[DriveSignal] = {
-      toClosedDrive(forwardSpeed.map(f => UnicycleSignal(f, Percent(0))))
+    def forwardOpenLoop(forwardSpeed: Signal[Dimensionless]): PeriodicSignal[DriveSignal] = {
+      parentClosedLoop(forwardSpeed.map(f => UnicycleSignal(f, Percent(0))))
     }
 
-    def turnOpen(turnSpeed: Signal[Dimensionless]): PeriodicSignal[DriveSignal] = {
-      toClosedDrive(turnSpeed.map(t => UnicycleSignal(Percent(0), t)))
+    def turnOpenLoop(turnSpeed: Signal[Dimensionless]): PeriodicSignal[DriveSignal] = {
+      parentClosedLoop(turnSpeed.map(t => UnicycleSignal(Percent(0), t)))
     }
 
-    def velocity(target: SignalLike[UnicycleVelocity]): PeriodicSignal[UnicycleSignal] = {
+    def velocityControl(target: SignalLike[UnicycleVelocity])
+                       (implicit hardware: DrivetrainHardware,
+                        props: DrivetrainProperties): PeriodicSignal[UnicycleSignal] = {
+      import hardware._, props._
+
       val forwardControl = PIDF.pidf(forwardVelocity, target.map(_.forward).toPeriodic, forwardControlGains)
       val turnControl = PIDF.pidf(turnVelocity, target.map(_.turn).toPeriodic, turnControlGains)
 
       forwardControl.zip(turnControl).map(s => UnicycleSignal(s._1, s._2))
     }
 
-    def expectedVelocity(unicycle: SignalLike[UnicycleSignal]): PeriodicSignal[UnicycleSignal] = {
-      velocity(unicycle.map(s => UnicycleVelocity(
+    def closedLoopControl(unicycle: SignalLike[UnicycleSignal])
+                         (implicit hardware: DrivetrainHardware,
+                          props: DrivetrainProperties): PeriodicSignal[UnicycleSignal] = {
+      import props._
+
+      velocityControl(unicycle.map(s => UnicycleVelocity(
```

Comment:
Here we want to take percentages, convert them to velocities and then pass it to velocityControl, which is exactly what you described with `UnicycleVelocity` as a parameter.